packages/terra-functional-testing/src/services/wdio-visual-regression-service/VisualRegressionLauncher.js

```diff
@@ -95,127 +75,41 @@ class VisualRegressionLauncher {
-   * @param {[Object]} capabilities - list of capabilities details
+   * Command wrapper to setup the command with the correct context values defined from the global
+   * browser instance.
+   *
+   * @param {object} browser - global wdio browser instance
+   * @param {function} command - the test command that should be executed
    */
-  async onComplete(_exitCode, _config, _capabilities) {
-    await this.runHook('onComplete');
-  }
-
-  // eslint-disable-next-line consistent-return
-  async runHook(hookName, ...args) {
-    if (typeof this.compare[hookName] === 'function') {
-      // eslint-disable-next-line no-return-await
-      return await this.compare[hookName](...args);
-    }
-  }
-
-  validateConfig(_config) {
-    if (!_.isPlainObject(this.options) || !_.has(this.options, 'compare')) {
-      throw new Error('Please provide a visualRegression configuration with a compare method in your wdio-conf.js!');
-    }
-  }
-
-  wrapCommand(browser, type, command) {
-    const baseContext = {
-      type,
-      browser: this.context.browser,
-      desiredCapabilities: this.context.desiredCapabilities,
-    };
-
-    const runHook = this.runHook.bind(this);
-
-    const getTestDetails = () => this.getTestDetails();
-
-    const resolutionKeyPlural = browser.isMobile ? 'orientations' : 'viewports';
-    const resolutionMap = browser.isMobile ? mapOrientations : mapViewports;
-
-    const viewportChangePauseDefault = this.viewportChangePause;
-    const resolutionDefault = browser.isMobile ? this.orientations : this.viewports;
-
-    return async function async(...args) {
-      const url = await browser.getUrl();
-
-      const elementSelector = type === 'element' ? args[0] : undefined;
-      const options = _.isPlainObject(args[args.length - 1]) ? args[args.length - 1] : {};
-
-      const { exclude, hide, remove } = options;
-
-      const resolutions = _.get(options, resolutionKeyPlural, resolutionDefault);
-      const viewportChangePause = _.get(options, 'viewportChangePause', viewportChangePauseDefault);
-
-      const results = await resolutionMap(
-        browser,
-        viewportChangePause,
-        resolutions,
-        // eslint-disable-next-line prefer-arrow-callback
-        async function takeScreenshot(currentFormFactor) {
-          const meta = _.pickBy(
-            {
-              url,
-              element: elementSelector,
-              exclude,
-              hide,
-              remove,
-              currentFormFactor,
-            },
-            _.identity,
-          );
-
-          const screenshotContext = {
-            ...baseContext,
-            ...getTestDetails(),
-            meta,
-            options,
-          };
-
-          const screenshotContextCleaned = _.pickBy(screenshotContext, _.identity);
-
-          await runHook('beforeScreenshot', screenshotContextCleaned);
-
-          const base64Screenshot = await command(browser, ...args);
+  wrapCommand(browser, command) {
```

Comment:
This looks pretty messy, but this slims what is passed as the screenshot context and remove the ability to loop changing viewport sizes and taking a screenshot.

This still supports mobile orientations.